dotCMS/src/main/java/com/dotcms/util/AssetExporterUtil.java

```diff
@@ -0,0 +1,183 @@
+        final String hostName = Try
+                .of(() -> APILocator.getHostAPI().findDefaultHost(APILocator.systemUser(), false).getHostname())
+                .getOrElse("dotcms");
+        return StringUtils.sanitizeFileName(hostName)
+                + "_assets_"
+                + DateUtil.EXPORTING_DATE_FORMAT.format(new Date())
+                + ".tar.gz";
+    }
+
+    /**
+     * Builds the tar.gz file containing the assets.
+     * 
+     * @param out output stream
+     */
+     public static void exportAssets(final OutputStream out) throws IOException {
+         synchronized (AssetExporterUtil.class) {
+             try (final TarArchiveOutputStream taos =
+                          new TarArchiveOutputStream(new GZIPOutputStream(new BufferedOutputStream(out)))) {
+                 taos.setBigNumberMode(TarArchiveOutputStream.BIGNUMBER_STAR);
+                 taos.setLongFileMode(TarArchiveOutputStream.LONGFILE_GNU);
+
+                 FileUtil.listFilesRecursively(PARENT.get(), FILE_FILTER)
+                         .stream()
+                         .filter(File::isFile)
+                         .forEach(file -> {
+                             final String relativeFilePath = file.getPath().replace(PARENT.get().getPath(), "assets");
+                             Logger.info(
+                                     AssetExporterUtil.class,
+                                     String.format(
+                                             "Adding file %s (%s) to zip output stream",
+                                             file.getName(),
+                                             relativeFilePath));
+                             final TarArchiveEntry tarEntry = new TarArchiveEntry(file, relativeFilePath);
+                             tarEntry.setSize(file.length());
+                             try {
+                                 taos.putArchiveEntry(tarEntry);
+                                 try (InputStream in = Files.newInputStream(file.toPath())) {
+                                     taos.write(IOUtils.toByteArray(in));
+                                 }
+                                 taos.closeArchiveEntry();
+                             } catch (Exception e) {
+                                 Logger.error(AssetExporterUtil.class, "Error generating assets zip file", e);
+                                 throw new DotRuntimeException(e);
+                             }
+                         });
+             }
+         }
+    }
+
+    // TODO: verify if this is still the case
+    /**
+     * Unused at this time, intended to load a list of live/working inodes to check against so old
+     * versions do not get exported.
+     * 
+     * @param version version
+     * @param lastCursorId last cursor id
+     * @return loaded inodes
+     */
+    @CloseDBIfOpened
+    static Tuple2<String, Set<String>> loadUpInodes(final EXPORT_VERSION version, final String lastCursorId) {
```

Comment:
I would say we should create a ExporterAPI and ExporterFactory (DAO) to encapsulate this thing

Reply:
Removed since it's not used. Actual code remains in plugin repo.